### PR TITLE
Support for kubernetes v1.23.0 in e2e tests

### LINKED ISSUE #1151: Remove 1.19 from machine-controller e2e tests
KKP no longer supports 1.19 on `master` as it's EOL, we should remove 1.19 from our e2e tests in `machine-controller` as well. At the moment, we're testing four Kubernetes minor versions, so that's quite a number of VMs we are spinning up.

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
Support for Kubernetes v1.23.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #1151 #1157

**Special notes for your reviewer**:

Supported k8s versions have been synced with what we **will** support from the dashboard/API in KKP:
```
* v1.20.14
* v1.21.8
* v1.22.5
* v1.23.0
```

**Optional Release Note**:

```release-note
* Support for v1.23.0
* Remove tests for v1.19.15
* Bump hetzner terraform provider to 1.31.0
```

### CODE REVIEW

File: examples/scaleway-machinedeployment.yaml [hunk 1]
```diff
@@ -61,4 +61,4 @@ spec:
           operatingSystemSpec:
             disableAutoUpdate: true
       versions:
-        kubelet: 1.21.5
+        kubelet: 1.23.0
```
Comment (embik) [inline examples/scaleway-machinedeployment.yaml:64]: Minor nit, let's make this `1.22.5` as with the other examples.

**Review by embik — CHANGES_REQUESTED** [review]

File: examples/scaleway-machinedeployment.yaml (hunk 1 above)
Comment (ahmedwaleedmalik) [reply examples/scaleway-machinedeployment.yaml:64]: Good catch :grin:

**Review by embik — APPROVED** [review]

**Pull request merged**